hotsos/core/ycheck/engine/properties/requires/types/binary.py

```diff
@@ -0,0 +1,59 @@
+import copy
+from functools import cached_property
+
+from hotsos.core.log import log
+from hotsos.core.ycheck.engine.properties.requires import (
+    intercept_exception,
+    YRequirementTypeBase,
+)
+from hotsos.core.ycheck.engine.properties.requires.types.apt import \
+    APTCheckItems
+
+
+class BinCheckItems(APTCheckItems):
+
+    def __init__(self, *args, bin_handler=None, **kwargs):
+        self.bin_handler = bin_handler
+        super().__init__(*args, **kwargs)
+
+    @cached_property
+    def packaging_helper(self):
+        return self.bin_handler()
+
+
+class YRequirementTypeBinary(YRequirementTypeBase):
+    _override_keys = ['binary']
+    _overrride_autoregister = True
+
+    @property
+    @intercept_exception
+    def _result(self):
+        _content = copy.deepcopy(self.content)
+        del _content['handler']
+        items = BinCheckItems(_content, bin_handler=self.get_cls(self.handler))
+
+        _result = True
+        # bail on first fail i.e. if any not installed
+        if not items.not_installed:
+            for binary, versions in items:
+                log.debug("binary %s installed=%s", binary, _result)
+                if not versions:
+                    continue
+
+                _result = items.package_version_within_ranges(binary, versions)
```

Comment:
I guess this will do the right thing and you don't need a binary_version_within_ranges() type function?

Reply:
Correct, as long as the versions follow apt/dpkg logic it works. If this doesnt work at some later point we can make it overridable but for now we can re-use the apt logic.